### PR TITLE
Fix by type always uses temp view

### LINKED ISSUE #45: Documents filter by type method should exclude docs by default
Currently fetching documents by type includes documents in result. Given that filtering by type does not inherently imply a need to fetch associated documents, the default behaviour should be to exclude docs.

### LINKED ISSUE #48: Document byType filter always uses temporary views
The API (both deprecated and new) for filtering documents by type using a permanent view uses the couch temporary views API (`/{db_name}/_temp_view`) under the hood, instead of the appropriate couch view APIs (`{db_name}/_design/{design_name}/_view/{view_name}`). This of course is not the desired behaviour and is inefficient. The APIs for filtering using a permanent view might have to change instead to take a permanent view name.

### PR DESCRIPTION
**Pull request by ermyas**
Fixes #48 Fixes #45

### CODE REVIEW

File: src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala [hunk 1]
```diff
@@ -96,18 +96,41 @@ BT <: DocType] private(
...
+
+  @deprecated("Use byTypeUsingTemporaryView(mappedType: MappedDocType) instead", "0.7.2")
   def byTypeUsingTemporaryView[D: R]:
   GetManyDocumentsQueryBuilder[IncludeDocs[D], AM, ForDocType[(String, String), String, D]] = {
     set(params, ids, Some(tempTypeFilterView))
   }
 
-  def inclusiveEnd(
-      inclusiveEnd: Boolean = true): GetManyDocumentsQueryBuilder[ID, AM, BT] = {
+  def byTypeUsingTemporaryView[D: R](mappedType: MappedDocType):
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala]: Looks like `D` is not required here.

File: src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala [hunk 2]
```diff
@@ -96,18 +96,41 @@ BT <: DocType] private(
...
+  ViewQueryBuilder[(String, String), String, ID, MapOnly] = {
+    new ViewQueryBuilder[(String, String), String, ID, MapOnly](
```
Comment (beloglazov) [inline src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala:128]: Type parameters get inferred?

**Comment by beloglazov** [comment]
Shall we change the version to 0.7.2-SNAPSHOT?

File: src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala (hunk 2 above)
Comment (ermyas) [inline src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala:128]: no can't be inferred here

**Comment by beloglazov** [comment]
LGTM

**Pull request merged**